src/agoradatatools/gx.py

```diff
@@ -148,43 +150,59 @@ def convert_nested_columns_to_json(
             df[column] = df[column].apply(json.dumps)
         return df
 
-    def get_failed_expectations(self, checkpoint_result: CheckpointResult) -> str:
-        """Gets the failed expectations from a CheckpointResult and returns them as a formatted string
+    def set_warnings_and_failures(self, checkpoint_result: CheckpointResult) -> None:
+        """Sets class attributes for warnings and failures
 
         Args:
             checkpoint_result (CheckpointResult): CheckpointResult object
-
-        Returns:
-            fail_message (str): String with information on which fields and expectations failed
         """
+        warning_dict = {self.expectation_suite_name: {}}
         fail_dict = {self.expectation_suite_name: {}}
         expectation_results = checkpoint_result.list_validation_results()[0]["results"]
+
         for result in expectation_results:
-            if not result["success"]:
-                column = result["expectation_config"]["kwargs"]["column"]
-                failed_expectation = result["expectation_config"]["expectation_type"]
-                if not fail_dict[self.expectation_suite_name].get(column, None):
-                    fail_dict[self.expectation_suite_name][column] = []
-                fail_dict[self.expectation_suite_name][column].append(
-                    failed_expectation
+            column = result["expectation_config"]["kwargs"].get(
+                "column",
+                "/".join(result["expectation_config"]["kwargs"].get("column_list", [])),
+            )
```

Comment:
This should help us to report out failures for expectations that are based on more than one column